components/browser/toolbar/src/main/java/mozilla/components/browser/toolbar/BrowserToolbar.kt

```diff
@@ -71,445 +57,91 @@ class BrowserToolbar @JvmOverloads constructor(
-        }
-
-    /**
-     * Set progress bar to be at the top of the toolbar. It's on bottom by default.
-     */
-    var progressBarGravity: Int
-        get() = displayToolbar.progressBarGravity
-        set(value) {
-            displayToolbar.progressBarGravity = value
-        }
-
-    /**
-     * Sets the colour of the text for title displayed in the toolbar.
-     */
-    var titleColor: Int
-        get() = displayToolbar.urlView.currentTextColor
-        set(value) {
-            displayToolbar.titleView.setTextColor(value)
-        }
-
-    /**
-     * Sets the colour of the text for the URL/search term displayed in the toolbar.
-     */
-    var textColor: Int
-        get() = displayToolbar.urlView.currentTextColor
-        set(value) {
-            displayToolbar.urlView.setTextColor(value)
-            editToolbar.urlView.setTextColor(value)
-        }
-
-    /**
-     * Sets the colour of the tracking protection icon.
-     */
-    var trackingProtectionColor: Int
-        get() = displayToolbar.trackingProtectionViewColor
-        set(value) {
-            displayToolbar.trackingProtectionViewColor = value
-        }
+    private var state: State = State.DISPLAY
+    private var searchTerms: String = ""
+    private var urlCommitListener: ((String) -> Boolean)? = null
 
     /**
-     * Sets the different icons that the tracking protection icon could has depending of its
-     * [Toolbar.siteTrackingProtection]
-     * @param iconOnNoTrackersBlocked icon for when the site is on the state
-     * [Toolbar.SiteTrackingProtection.ON_NO_TRACKERS_BLOCKED]
-     * @param iconOnTrackersBlocked icon for when the site is on the state
-     * [Toolbar.SiteTrackingProtection.ON_TRACKERS_BLOCKED]
-     * @param iconDisabledForSite icon for when the site is on the state
-     * [Toolbar.SiteTrackingProtection.OFF_FOR_A_SITE]
+     * Toolbar in "display mode".
      */
-    fun setTrackingProtectionIcons(
-        iconOnNoTrackersBlocked: Drawable = requireNotNull(
-            context.getDrawable(
-                TrackingProtectionIconView.DEFAULT_ICON_ON_NO_TRACKERS_BLOCKED
-            )
-        ),
-        iconOnTrackersBlocked: Drawable = requireNotNull(
-            context.getDrawable(
-                TrackingProtectionIconView.DEFAULT_ICON_ON_TRACKERS_BLOCKED
-            )
-        ),
-        iconDisabledForSite: Drawable = requireNotNull(
-            context.getDrawable(
-                TrackingProtectionIconView.DEFAULT_ICON_OFF_FOR_A_SITE
-            )
-        )
-    ) {
-        displayToolbar.setTrackingProtectionIcons(
-            iconOnNoTrackersBlocked,
-            iconOnTrackersBlocked,
-            iconDisabledForSite
+    var display = DisplayToolbar(
+        context,
+        this,
+        LayoutInflater.from(context).inflate(
+            R.layout.mozac_browser_toolbar_displaytoolbar,
+            this,
+            false
         )
-    }
+    )
+    @VisibleForTesting(otherwise = PRIVATE) internal set
```

Comment:
I'm not fully sure what you mean. Did you see that I added it to the setter of the attribute above (`internal set`)?

Reply:
I'm a bit confused. I thought you were trying to add the annotation to the `edit` property.

What I meant is that normally we put annotations close to the code, like this.
![image](https://user-images.githubusercontent.com/773158/67010751-4134ad80-f0bc-11e9-84dd-c27c21fa9ac4.png)

And now the annotation is far from the code and I'm not sure to which property it belongs to.
` edit setter` or  `display setter`?

![image](https://user-images.githubusercontent.com/773158/67010557-de431680-f0bb-11e9-98ff-3325552ba645.png)

Reply:
Yeah, but I am not setting the annotation on the property but only on the "setter"! 

See: https://kotlinlang.org/docs/reference/properties.html#getters-and-setters

The syntax is:

```kotlin
var <propertyName>[: <PropertyType>] [= <property_initializer>]
    [<getter>]
    [<setter>]
```

In your example above you are now changing the visiblity of the property **above** and adding the annotation to it. You are **not** setting it to anything on `edit`.

Reply:
Got it!

Reply:
Maybe its easier to read if I write it as:

```Kotlin
var edit = EditToolbar(
...
)
@VisibleForTesting(otherwise = PRIVATE)
internal set
```

What do you think?

Reply:
Yes, it makes it a bit easier. What made it a bit hard to understand for me, was that I'm used to see annotations on top of the code that they get applied, no below :)